dive_core/common/dive_annotation_processor.h

```diff
@@ -0,0 +1,109 @@
+/*
+ Copyright 2025 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include "dive_core/stl_replacement.h"
+#include "third_party/gfxreconstruct/framework/decode/annotation_handler.h"
+#include "util/defines.h"
+#include "util/platform.h"
+
+struct ApiCallInfo;
+
+/// Manages writing
+class DiveAnnotationProcessor : public gfxrecon::decode::AnnotationHandler
+{
+public:
+    struct VulkanCommandInfo
+    {
+    public:
+        VulkanCommandInfo(const std::string& name, uint32_t index)
+        {
+            m_name = name;
+            m_index = index;
+        }
+        VulkanCommandInfo(const gfxrecon::util::DiveFunctionData& data)
+        {
+            m_name = data.GetFunctionName();
+            m_index = data.GetCmdBufferIndex();
+            m_args = data.GetArgs();
+        }
+        const std::string&            GetVkCmdName() const { return m_name; }
+        uint32_t                      GetVkCmdIndex() const { return m_index; }
+        void                          SetCmdCount(uint32_t cmd_count) { m_cmd_count = cmd_count; }
+        uint32_t                      GetCmdCount() const { return m_cmd_count; }
+        const nlohmann::ordered_json& GetArgs() const { return m_args; }
+
+    private:
+        std::string            m_name;
```

Comment:
generally speaking, it is better to sort member declarations based on the size of their types (from largest to smallest) to avoid potential extra padding